src/amount.rs

```diff
@@ -82,4 +116,15 @@ mod tests {
     fn invalid_amount() {
         assert!(amount("10 chf").is_err());
     }
+
+    #[rstest]
+    fn valid_currency(#[values("CHF", "X-A", "A", "AB")] input: &str) {
```

Comment:
since you are at it, can you also add the cases of `_`, `.` and `'`?

That would be awesome :-)